packages/approval-controller/src/ApprovalController.ts

```diff
@@ -267,23 +323,48 @@ export class ApprovalController extends BaseControllerV2<
    * @param opts.requestData - Additional data associated with the request,
    * @param opts.requestState - Additional state associated with the request,
    * if any.
-   * @returns The approval promise.
+   * @returns The approval promise resolving to an AddResult object.
+   */
+  addAndShowApprovalRequest(
+    opts: AddApprovalOptions & { expectsResult: true },
+  ): Promise<AddResult>;
```

Comment:
TIL this absolutely works:

https://www.typescriptlang.org/play?#code/C4TwDgpgBAggJnGYwCcD2A3AhgGwPJjACWaAdgM5QC8UA3gFBRNQQAekAxsOQEoTkBXHMAD8ALigAjNGhwQspADT0AvgG569UJFgI+g4dTqNmAMxkTywFEVIBzZes1wIHHFhTRTA0lxKkoLAQACjRCcgl4RGR0bHxCf0oAMjoWdldufSFgCWsBaBUASgkABXQAWyJyCAAeKKzhAD4NFzcPLx8-MkCQsO5IhCRUTFwCYjJyYqgytErqmp8Aa1I0AHdSZs0guGDaIoA6YAALCFJguPzqRqgOCdkIfZw0O3PcfP3zNELCjW3dtM4mX42Qkplw1SgB2Op1eOEuVGutwo90ez1h70+31+IVoAIyvGBwlyKEuUJOZwu0ARNzuclRL0pHxkWKAA